### PR TITLE
[IOTDB-1396] Restructure logical query operator

### PR DESCRIPTION
**Pull request by Alima777**
The design doc is in https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=181306231

### CODE REVIEW

**Review by SteveYurongSu — COMMENTED** [review]
Not done my review yet.

File: server/src/main/java/org/apache/iotdb/db/qp/utils/WildcardsRemover.java [hunk 1]
```diff
@@ -35,42 +36,36 @@
 import java.util.Collections;
 import java.util.List;
 
-/** Removes wildcards (applying memory control and slimit/soffset control) */
```
[the remark this answers is not in the document]
Comment (Alima777) [reply server/src/main/java/org/apache/iotdb/db/qp/utils/WildcardsRemover.java:38]: Sorry, the branch conflict is sooooo complex that I delete it incorrectly.

**Comment by SteveYurongSu** [comment]
image

File: server/src/main/java/org/apache/iotdb/db/qp/logical/crud/AggregationQueryOperator.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+public class AggregationQueryOperator extends QueryOperator {
+
+  public static final String ERROR_MESSAGE1 =
+      "Common queries and aggregated queries are not allowed to appear at the same time";
+
+  public AggregationQueryOperator() {}
+
+  public AggregationQueryOperator(QueryOperator queryOperator) {
+    this.selectComponent = queryOperator.getSelectComponent();
```
Comment (SteveYurongSu) [inline server/src/main/java/org/apache/iotdb/db/qp/logical/crud/AggregationQueryOperator.java:35]: implement 
```java
public QueryOperator(QueryOperator queryOperator)
```
and use    
```java
super(queryOperator)
```
here?

File: server/src/main/java/org/apache/iotdb/db/qp/logical/crud/LastQueryOperator.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import org.apache.iotdb.db.exception.query.LogicalOperatorException;
+import org.apache.iotdb.db.query.expression.Expression;
+import org.apache.iotdb.db.query.expression.ResultColumn;
+import org.apache.iotdb.db.query.expression.unary.TimeSeriesOperand;
+
+public class LastQueryOperator extends QueryOperator {
+
+  public LastQueryOperator() {}
+
+  public LastQueryOperator(QueryOperator queryOperator) {
```
Comment (SteveYurongSu) [inline server/src/main/java/org/apache/iotdb/db/qp/logical/crud/LastQueryOperator.java:31]: implement 
```java
public QueryOperator(QueryOperator queryOperator)
```
and use    
```java
super(queryOperator)
```
here?

File: server/src/main/java/org/apache/iotdb/db/qp/logical/crud/UDFQueryOperator.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ * under the License.
+ */
+
+package org.apache.iotdb.db.qp.logical.crud;
+
+public class UDFQueryOperator extends QueryOperator {
+
+  public UDFQueryOperator() {}
+
+  public UDFQueryOperator(QueryOperator queryOperator) {
```
Comment (SteveYurongSu) [inline server/src/main/java/org/apache/iotdb/db/qp/logical/crud/UDFQueryOperator.java:26]: implement 
```java
public QueryOperator(QueryOperator queryOperator)
```
and use    
```java
super(queryOperator)
```
here?

File: server/src/main/java/org/apache/iotdb/db/qp/logical/crud/AggregationQueryOperator.java (hunk 2 above)
Comment (Alima777) [reply server/src/main/java/org/apache/iotdb/db/qp/logical/crud/AggregationQueryOperator.java:35]: Good suggestion~

File: server/src/main/java/org/apache/iotdb/db/qp/logical/crud/FilterOperator.java [hunk 5]
```diff
@@ -64,9 +64,6 @@
   public FilterOperator(int tokenType) {
     super(tokenType);
     operatorType = OperatorType.FILTER;
```
[the remark this answers is not in the document]
Comment (Alima777) [reply server/src/main/java/org/apache/iotdb/db/qp/logical/crud/FilterOperator.java:66]: I think so! Maybe we can try to remove `operatorType` in another PR.

**Review by SteveYurongSu — APPROVED** [review]

File: server/src/main/java/org/apache/iotdb/db/qp/logical/crud/FilterOperator.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (Alima777) [reply server/src/main/java/org/apache/iotdb/db/qp/logical/crud/FilterOperator.java]: I removed the `filteroperator` from the inheritance of `operator`. Later, I will try to remove`tokenIntType` or `operatorType`, just leave one of them.

**Review by SteveYurongSu — APPROVED** [review]

**Pull request merged**